src/vm/amd64/pinvokestubs.S

```diff
@@ -130,24 +133,92 @@ NESTED_END VarargPInvokeGenILStub, _TEXT
 
 //
 // IN:
-// InlinedCallFrame (rcx) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
+// InlinedCallFrame (rdi) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
 //                          before actual InlinedCallFrame data)
 //
 //
-LEAF_ENTRY JIT_PInvokeBegin, _TEXT
-        // Not yet supported
-        int 3
+NESTED_ENTRY JIT_PInvokeBegin, _TEXT, NoHandler
+
+        alloc_stack         MIN_SIZE
+        save_reg_postrsp    r14, (MIN_SIZE + 0x08)
+        END_PROLOGUE
+
+        PREPARE_EXTERNAL_VAR s_gsCookie, rax
+        mov             rax, qword ptr [rax]
+        mov             qword ptr [rdi], rax
+        add             rdi, SIZEOF_GSCookie
+
+        // set first slot to the value of InlinedCallFrame vftable (checked by runtime code)
+#ifdef __APPLE__
+        mov             rax, qword ptr [rip + __ZTV16InlinedCallFrame@GOTPCREL]
```

Comment:
You can use C_FUNC(_ZTV16InlinedCallFrame@GOTPCREL) to unify the APPLE / non-APPLE code here.

Reply:
I couldn't really unify it. My latest unification attempt made the build succeed on both OSX and Ubuntu, but the CI failed for ubuntu due to umresolved symbol issues...